node/services/service.go

```diff
@@ -167,21 +175,38 @@ func LightAvailability(
 	lc fx.Lifecycle,
 	bServ blockservice.BlockService,
 	ds datastore.Batching,
+	r routing.ContentRouting,
+	h host.Host,
 ) share.Availability {
-	la := share.NewLightAvailability(bServ)
+	la := share.NewLightAvailability(bServ, routingdisc.NewRoutingDiscovery(r), h)
 	ca := share.NewCacheAvailability(la, ds)
 	lc.Append(fx.Hook{
-		OnStop: ca.Close,
+		OnStart: la.Start,
+		OnStop: func(ctx context.Context) error {
+			_ = la.Stop(ctx)
```

Comment:
I know that we don't return an error there, but we might at some point and when we do this, we can forget the error here, so let's add a check here. Same with Full

Reply:
Ideally, we should do this instead https://github.com/celestiaorg/celestia-node/issues/879.
Maybe this fix could be part of that issue.

Reply:
added multierr for now